packages/@aws-cdk-testing/cli-integ/resources/cdk-apps/app/app.js

```diff
@@ -417,6 +417,36 @@ class LambdaHotswapStack extends cdk.Stack {
   }
 }
 
+class DetailedStatusStack extends cdk.Stack{
+  constructor(parent, id, props) {
+    super(parent, id, props);
+
+    const vpc = new ec2.Vpc(this, 'Vpc', {
+      natGateways: 0,
+      maxAzs: 1,
+      subnetConfiguration: [
+        {
+          cidrMask: 24,
+          name: 'Public',
+          subnetType: ec2.SubnetType.PUBLIC,
+        },
+      ],
+    });
+
```

Comment:
A high level question -- what if a resource fails deploying after CONFIGURATION_COMPLETE status is emitted? Do we handle that case? Is that a possible case? If it is a possible case, could we have an integration test for that?

Reply:
Will discuss this with the CFN team.

Reply:
My thinking is,  if that happens, we should fail with exit status 1. One idea, without much thinking about how this would work, is to poll the statuses of all the stacks once the final stack has entered CONFIG_COMPLETE. So then it's just that the final stack of the deployment is not getting the speedup... which I think is fine since you then get the speed up on all the intermediate stacks.

I am considering if there is a case where a customer kicks off some sort of production thing as soon as the customer get a success signal from CDK deploy? If so, then we'd want to make sure that everything did successfully provision... unless CloudFormation handles that, somehow. I'm not too familiar with this new optimistic stabilization feature from CloudFormation.

Reply:
We probably should figure out, let's say if we have 2 stacks to deploy with Stack2 depends on Stack1:

Stack1 enters `CC` state -> CLI starts deploying Stack2 -> Stack2 enters `CC` state -> CLI exit 0 but Stack1 ended up with final status failed. Now we get boost from Stack1 and Stack2 but Stack1 is actually failed. What should we do?

Reply:
My preconception (which could be wrong) is that stack 2 would fail if stack 1 eventually failed. But if the scenario you're describing is true, then that biases me in favor of including a warning about this not being for production, and that would be all we'd do for addressing the failure case. So, similar to how we advise users to not use hotswap in production, we'd advise users to not use this flag in production. What do you think of that? I personally feel satisfied with that.

Reply:
I'll check with the CFN team:

-  Will stack2 fail when stack1 fails with `CC` status
a. if stack2 would fail then we just need to check the status of the last stack
b. if stack2 would NOT fail we will end up with a failed stack1 and a deployed stack2.

Reply:
Yes! I agree with @bergjaak to warn customer to not use the flag in production. This is what CFN official user guide mentions. 

Stack 2 will not fail automatically. unless CDK implement a pulling mechanism that read the state an fail the stack based on the stack 1 state.